### PR TITLE
Wrote Unit Tests for JsonProject

### LINKED ISSUE #152: Write Unit Tests for Output Class JsonProject
Similarly to how unit tests are written for JsonTask.

### PR DESCRIPTION
**Pull request by Lumi3011**
Fixes #152 
Finished Unit Test for Output Class JsonProject

### CODE REVIEW

File: src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java [hunk 1]
```diff
@@ -0,0 +1,578 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (amihaiemil) [inline src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java:5]: @Lumi3011 Can you copy out licence header here? Just look in any other file and copy it here instead of this placeholder.

File: src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+import javax.json.JsonObject;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.hamcrest.Matchers;
+import org.hamcrest.MatcherAssert;
+
+
+
+/**
+ * @author Lumi
```
Comment (amihaiemil) [inline src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java:25]: @Lumi3011 Can you say: "Unit tests for {@link JsonProject}."?

File: src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+                ).build())
+        );       
+    }
+    
+    /**
+     * JsonProject has null repoFullName
+     */
+    
+    @Test
+    public void hasNullRepoFullName(){
```
Comment (amihaiemil) [inline src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java:310]: @Lumi3011 This test can be removed. It's logically identical to "hasRepoFullName", it just works with the ``"null"`` String.

File: src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java [hunk 4]
```diff
@@ -0,0 +1,578 @@
...
+        );   
+}
+    
+    /**
+     * JsonProject has null Provider
+     */
+    
+    
+    @Test
+    public void hasNullProvider(){
```
Comment (amihaiemil) [inline src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java:361]: @Lumi3011 This one can also be removed, it's logically identical to ``hasProvider()``

File: src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java [hunk 5]
```diff
@@ -0,0 +1,578 @@
...
+            Matchers.equalTo("null")
+        );       
+    }
+/**
+     * JsonProject has null Owner
+     */
+    
+    
+    @Test
+    public void hasNullOwner(){
```
Comment (amihaiemil) [inline src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java:411]: @Lumi3011 same here, remove it please :D

File: src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java [hunk 6]
```diff
@@ -0,0 +1,578 @@
...
+        
+    }
+     
+    /**
+     * JsonProject has null Wallet
+     */
+    
+    
+    @Test
+    public void hasNullWallet(){
```
Comment (amihaiemil) [inline src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java:526]: @Lumi3011 This one as well, removed

File: src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java [hunk 7]
```diff
@@ -0,0 +1,578 @@
...
+        );            
+        
+    }
+     /**
+     * JsonProject has null Manager
+     */
+    
+    
+    @Test
+    public void hasNullManager(){
```
Comment (amihaiemil) [inline src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java:462]: @Lumi3011 remove it too please

File: src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java [hunk 8]
```diff
@@ -0,0 +1,578 @@
...
+                .build())
+        );       
+        
+    }
+    
+    /**
+     * JsonProject has Wallet
+     */
+    
+
```
Comment (amihaiemil) [inline src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java:250]: @Lumi3011 there should be no empty lines between JavaDocs and methods/tests. Can you please remove the empty lines between all javadocs and their tests?

**Review by amihaiemil — CHANGES_REQUESTED** [review]
@Lumi3011 Please see my comments.

When you want to test something with ``null``, you should use the null reference, not the ``"null"`` String, which is just a String line any other.

In our case, we don't need tests for ``null``, because we know that ``JSON-P`` does not support adding null values when building Json.

**Comment by Lumi3011** [comment]
So I should delete all the verifications for null, or just the ones that are String?

**Comment by amihaiemil** [comment]
@Lumi3011 Just delete the ``hasNull*`` tests, they are not testing the same thing as others :)

**Comment by Lumi3011** [comment]
Ok :D

**Comment by Lumi3011** [comment]
@amihaiemil I have done all the modifications on this branch, hope is ok now.

**Comment by amihaiemil** [comment]
@Lumi3011 great, thanks!

**Comment by amihaiemil** [comment]
@rultor merge it please
